src/ResourceTemplate/Segment.php

```diff
@@ -200,6 +204,10 @@ private static function isValidBinding($binding)
      */
     private static function isValidDoubleWildcardBinding($binding)
     {
+        if ($binding === null) {
+            return false;
+        }
```

Comment:
in these two cases as well, I am sure why we are accepting `null` as a valid parameter option. It would be better to just typehint `$binding` as a string.